qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -107,9 +146,45 @@ def _substitution_checks(self, dag, old_run, new_circ, new_basis):
             for g in old_run
         )
 
-        if rewriteable_and_in_basis_p and len(old_run) < len(new_circ):
-            # NOTE: This is short-circuited on calibrated gates, which we're timid about
-            #       reducing.
+        old_infidelity = 0
+        new_infidelity = 0
+        # look up physical qubit index in layout
+        # (this snippet is adapted from UnitarySynthesis._synth_natural_direction)
+        qubit = next(
+            (index for (index, qubit) in enumerate(dag.qubits) if qubit == old_run[0].qargs[0]),
+            None,
+        )
+
+        if self._backend_properties is not None:
+            try:
+                for op in old_run:
+                    if not isinstance(op, DAGOpNode):
+                        continue
+                    old_infidelity += self._backend_properties.gate_error(op.op.name, [qubit])
+            except BackendPropertyError:
+                old_infidelity += float("inf")
+
+            for instr, _, _ in new_circ.data:
+                try:
+                    new_infidelity += self._backend_properties.gate_error(instr.name, [qubit])
+                except BackendPropertyError:
+                    pass
+
+        # incorporate a possible trace distance from approximate synthesis
+        operator = old_run[0].op.to_matrix()
+        for gate in old_run[1:]:
+            operator = gate.op.to_matrix().dot(operator)
+        decomp_unitary = new_circ[0][0].to_matrix() if new_circ else np.eye(2)
+        for gate, _, _ in new_circ[1:]:
+            decomp_unitary = gate.to_matrix().dot(decomp_unitary)
+        trace_pairing = np.trace(decomp_unitary @ np.conj(operator.data).transpose(1, 0))
+        new_infidelity += (4 - abs(trace_pairing) ** 2) / 6
+
+        if rewriteable_and_in_basis_p and (
+            (old_infidelity + atol < new_infidelity)
+            or (abs(old_infidelity - new_infidelity) < atol and len(old_run) < len(new_circ))
+        ):
+            # NOTE: This is short-circuited on calibrated gates, which we're timid about reducing.
             warnings.warn(
```

Comment:
I'm ok with making this a warning again since this is supposed to fix the problem completely that led us to demoting to debug log.